core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceMethod.java

```diff
@@ -318,27 +322,68 @@ private HttpResponse convertException(ServiceRequestContext ctx, HttpRequest req
      * Creates a {@link Parameter} instance which describes a parameter of the annotated method. If the
      * parameter type is {@link Optional}, its actual argument's type is used.
      */
-    private static Parameter createOptionalSupportedParam(java.lang.reflect.Parameter parameterInfo,
+    private static Parameter createHttpComponentParameter(java.lang.reflect.Parameter parameterInfo,
                                                           ParameterType paramType, String paramValue) {
-        final Default aDefault = parameterInfo.getAnnotation(Default.class);
-        final boolean isOptionalType = parameterInfo.getType() == Optional.class;
+        assert paramType == ParameterType.PARAM || paramType == ParameterType.HEADER
+                : String.valueOf(paramType);
 
+        final Default aDefault = parameterInfo.getAnnotation(Default.class);
         // Set the default value to null if it was not specified.
         final String defaultValue = aDefault != null ? getSpecifiedValue(aDefault.value()).get() : null;
+
         final Class<?> type;
-        if (isOptionalType) {
+        final Class<?> wrapperType;
+
+        final Type parameterizedType = parameterInfo.getParameterizedType();
+        if (parameterizedType instanceof ParameterizedType) {
             try {
-                type = (Class<?>) ((ParameterizedType) parameterInfo.getParameterizedType())
-                        .getActualTypeArguments()[0];
-            } catch (Exception e) {
-                throw new IllegalArgumentException("Invalid optional parameter: " + parameterInfo.getName(), e);
+                type = (Class<?>) ((ParameterizedType) parameterizedType).getActualTypeArguments()[0];
+                wrapperType = validateWrapperAndElementType(paramType, parameterInfo.getType(), type);
+            } catch (Throwable cause) {
```

Comment:
Do we need to catch here? It seems like `validateWrapperAndElementType` throws same exception with the enough information

Reply:
You're right. Let me move the line to the out of the try-catch block. Thanks!